test/functional/api/highlight_spec.lua

```diff
@@ -158,3 +158,97 @@ describe('API: highlight',function()
     assert_alive()
   end)
 end)
+
+describe("API: set highlight", function()
+  local highlight_color = {
+    fg = tonumber('0xff0000'),
+    bg = tonumber('0x0032aa'),
+    ctermfg = 8,
+    ctermbg = 15,
+  }
+  local highlight1 = {
+    background = highlight_color.bg,
+    foreground = highlight_color.fg,
+    bold = true,
+    italic = true,
+  }
+  local highlight2_config = {
+    ctermbg = highlight_color.ctermbg,
+    ctermfg = highlight_color.ctermfg,
+    underline = true,
+    reverse = true,
+  }
+  local highlight2_result = {
+    background = highlight_color.ctermbg,
+    foreground = highlight_color.ctermfg,
+    underline = true,
+    reverse = true,
+  }
+  local highlight3_config = {
+    background = highlight_color.bg,
+    foreground = highlight_color.fg,
+    ctermbg = highlight_color.ctermbg,
+    ctermfg = highlight_color.ctermfg,
+    bold = true,
+    italic = true,
+    reverse = true,
+    undercurl = true,
+    underline = true,
+    cterm = {
+      italic = true,
+      reverse = true,
+      undercurl = true,
+    }
+  }
+  local highlight3_result_gui = {
+    background = highlight_color.bg,
+    foreground = highlight_color.fg,
+    bold = true,
+    italic = true,
+    reverse = true,
+    undercurl = true,
+    underline = true,
+  }
+  local highlight3_result_cterm = {
+    background = highlight_color.ctermbg,
+    foreground = highlight_color.ctermfg,
+    italic = true,
+    reverse = true,
+    undercurl = true,
+  }
+
```

Comment:
you probably want `before_each(clear)`

Reply:
I've added it but I'm not sure what it does . Does it call module.clear() in tests/functional/helpers.lua ?

Is it suppose to clear the namespace ?
Actually nvim_buf_clear_namespcae clears highlights of that namespace in  the buffer . But is there anything equavelent of :hi clear for namespace (removing highlight definations too not just applications )

Reply:
it restarts neovim so all state is cleared.